rust/kernel/thread.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::bindings;
+use crate::c_types;
+
+use alloc::boxed::Box;
+use core::ops::FnOnce;
+
+#[no_mangle]
+unsafe extern "C" fn rust_thread_func(data: *mut c_types::c_void) -> c_types::c_int {
+    Box::from_raw(data as *mut Box<dyn FnOnce()>)();
+    0
```

Comment:
We should probably have the lambda return a KernelResult, then extract the return error code from it and return it here.

Reply:
Done in the updated version